io/zenoh-transport/src/unicast/link.rs

```diff
@@ -391,3 +470,63 @@ async fn rx_task(
         rx_task_dgram(link, transport, lease, signal, rx_buffer_size).await
     }
 }
+
+#[cfg(feature = "transport_compression")]
+fn tx_compressed(
+    compression_enabled: bool,
+    bytes: &[u8],
+    compression_aux_buff: &mut Box<[u8]>,
+    link: &LinkUnicast,
+) -> ZResult<Vec<u8>> {
+    let mut buff = vec![];
+    if compression_enabled {
+        let zenoh_compress = ZenohCompress::default();
+        let compression_size = zenoh_compress
+            .write(&mut buff.writer(), (&bytes, compression_aux_buff))
+            .map_err(|e| zerror!("Compression error: {:?}", e))?;
+        let mut compressed_batch: Vec<u8> = vec![];
+        if link.is_streamed() {
+            let compression_size_u16: u16 = compression_size.try_into().map_err(|e| {
+                zerror!(
+                    "Compression error: unable to convert compression size into u16: {}",
+                    e
+                )
+            })?;
+            // including is compressed byte
+            let batch_size_le = (compression_size_u16 + 1).to_le_bytes();
+            compressed_batch.append(&mut batch_size_le.to_vec());
```

Comment:
This allocates everytime on the heap. I suggest to use slice copy instead of `push` and `append`.